addons/mail/models/mail_activity.py

```diff
@@ -355,16 +357,25 @@ def write(self, values):
                         self.env['bus.bus'].sendone(
                             (self._cr.dbname, 'res.partner', partner.id),
                             {'type': 'activity_updated', 'activity_deleted': True})
+        if values.get('activity_type_id'):
```

Comment:
Not sure we are writing res_model or res_id on activities, but to be complete, we would need call _set_activity_exception_type on record before write, in case the activity is switch to another one. Not sure it is a use case we need to support, not a big deal if we don't do that.

Reply:
@Xavier-Do , If we use before the write then activity_type_id will not be the right(it will be older one). and we used `activity_type_id` in `_set_activity_exception_type` Thanks

Reply:
I mean that if I write a res_id on an activity, the previously linked record will loose an activity. To be complete, we would need to do something like.
record_to_update= set()
if res_model  in values or res_id in values:
    for activity in self....
         record_to_update.append...
self.write....
if res_model  in values or res_id in values or activity_type_id in values:
    for activity in self....
         record_to_update.append...
for rec in record_to_update:
    rec.....

But since we are not switching activities from one record to another, it's maybe not the most necessary to do so.